### PR TITLE
Bug 1876700 - Add more test logs to CollectionRobot

### PR DESCRIPTION
**Pull request by AndiAJ**
Summary:

- As part of the Android test hardening, to ease the debugging process, I've added test logs to the `CollectionRobot`
- Also converted the private variables to functions so they don't get initialized
- Removed unused functions
- Use where applicable the functions from the `MatcherHelper` to avoid adding unnecessary logs 
- ❗There is [this](https://github.com/mozilla-mobile/firefox-android/pull/5313/files#diff-1f0f53606bb4c1c9a89ce422b9ac691fb4b2064d76f6e90cbf3f51ed96ba5d06R173) `verifySnackBarText` function which  actually doesn't assert anything, only waits for an item containing text.
   Will do a separate refactoring related to these snackbar verifications as we have a function in `TestHelper` which is only partially used in the project and have a couple more individual functions named as `verifySnackBarText`in different robots

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1876700

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/CollectionRobot.kt [hunk 1]
```diff
@@ -68,12 +75,14 @@ class CollectionRobot {
...
     fun saveTabsSelectedForCollection() {
-        mDevice.findObject(UiSelector().resourceId("$packageName:id/save_button")).click()
+        itemWithResId("$packageName:id/save_button").click()
+        Log.i(TAG, "saveTabsSelectedForCollection: Clicked \"Save\" button")
     }
 
     fun verifyTabSavedInCollection(title: String, visible: Boolean = true) {
         if (visible) {
             scrollToElementByText(title)
+            Log.i(TAG, "verifyTabSavedInCollection: Scrolled to: $title saved tab")
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/CollectionRobot.kt:85]: As this is a custom function, you could actually add a log inside the scroll function.

**Review by sv-ohorvath — APPROVED** [review]
Approved, with one single comment.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/CollectionRobot.kt (hunk 1 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/CollectionRobot.kt]: Done !

**Review by jjSDET — APPROVED** [review]
lgtm

**Pull request merged**